cmrf_views/src/Plugin/views/query/API4.php

```diff
@@ -0,0 +1,525 @@
+<?php namespace Drupal\cmrf_views\Plugin\views\query;
+
+use Drupal;
+use Drupal\cmrf_core\Call;
+use Drupal\cmrf_core\Core;
+use Drupal\cmrf_views\CMRFViewsResultRow;
+use Drupal\cmrf_views\Entity\CMRFDataset;
+use Drupal\views\Plugin\views\query\QueryPluginBase;
+use Drupal\views\ResultRow;
+use Drupal\views\ViewExecutable;
+use Drupal\views\ViewsData;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * CiviMRF CiviCRM Views query plugin which wraps calls to the API in order to
+ * expose the results to views.
+ *
+ * @ViewsQuery(
+ *   id = "civicrm_api4",
+ *   title = @Translation("CiviMRF CiviCRM APIv4"),
+ *   help = @Translation("Query against the CiviCRM APIv4.")
+ * )
+ */
+class API4 extends QueryPluginBase {
+
+  /**
+   * @var \Drupal\cmrf_core\Core
+   */
+  protected $core;
+
+  /**
+   * @var \Drupal\views\ViewsData
+   */
+  protected $viewsData;
+
+  /**
+   * API constructor.
+   *
+   * @param array                   $configuration
+   * @param                         $plugin_id
```

Comment:
Type hint is missing

Reply:
Feel free to add them, they're recommended but not required following Drupal coding standards for Drupal 9.